src/pages/signin/SignInPage.js

```diff
@@ -44,16 +46,32 @@ const propTypes = {
         // Error to display when there is a session error returned
         authToken: PropTypes.string,
     }),
+
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            validateCode: PropTypes.string,
+        }),
+    }),
 };
 
 const defaultProps = {
     account: {},
     session: {},
     credentials: {},
+    route: {
+        params: {},
+    },
 };
 
 class SignInPage extends Component {
     componentDidMount() {
+        // If we've redirected to the signin page with a validation code, validate the login
+        if (_.isMatch(this.props.route, {name: 'ValidateLogin'})) {
+            const accountID = Number(lodashGet(this.props.route, 'params.accountID', 0));
```

Comment:
NAB/fun tip, if we feel like letting `react-navigation` do this for us we can parse this param and cast it to a `Number` directly in the `linkingConfig.js` then we can always expect it to be a number.

https://reactnavigation.org/docs/configuring-links/#serializing-and-parsing-params